packages/wrangler/src/__tests__/patch-config.test.ts

```diff
@@ -0,0 +1,248 @@
+import { writeFileSync } from "node:fs";
+import dedent from "ts-dedent";
+import { experimental_patchConfig } from "../config/patch-config";
+import { runInTempDir } from "./helpers/run-in-tmp";
+import { writeWranglerConfig } from "./helpers/write-wrangler-config";
+import type { RawConfig } from "../config";
+
+type TestCase = {
+	name: string;
+	original: RawConfig;
+	patch: RawConfig;
+	expectedToml: string;
+	expectedJson: string;
+};
+const testCases: TestCase[] = [
```

Comment:
Could you add some tests for things like adding a D1 binding when only KV bindings exist, and adding a new field like compat date?

Reply:
a9e4d48ec0a781a6f8defb8245f2a263b9a32c33